src/de/mangatube/src/eu/kanade/tachiyomi/extension/de/mangatube/MangaTube.kt

```diff
@@ -40,146 +35,225 @@ class MangaTube : ParsedHttpSource() {
 
     override val supportsLatest = true
 
+    private val mangas: LinkedHashMap<String, Manga> = LinkedHashMap()
+
+    private val json = Json {
+        isLenient = true
+        ignoreUnknownKeys = true
+        allowSpecialFloatingPointValues = true
+        prettyPrint = true
+    }
+
     override val client: OkHttpClient = network.cloudflareClient.newBuilder()
         .connectTimeout(1, TimeUnit.MINUTES)
         .readTimeout(1, TimeUnit.MINUTES)
         .writeTimeout(1, TimeUnit.MINUTES)
         .build()
 
-    private val xhrHeaders: Headers = headersBuilder().add("Content-Type", "application/x-www-form-urlencoded; charset=UTF-8").build()
+    override fun imageUrlParse(response: Response): String = ""
+
+    override fun getMangaUrl(manga: SManga): String {
+        if (!manga.url.startsWith(baseUrl)) {
+            return "$baseUrl${manga.url}"
+        }
+        return manga.url
+    }
+
+    override fun getChapterUrl(chapter: SChapter): String {
+        return chapter.url
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$baseUrl/api/manga/search?page=$page&query=$query"
+
+        Log.d("MangaTube", "Search for: $query")
+        Log.d("MangaTube", "Url -> $url")
 
-    private val json: Json by injectLazy()
+        return GET(url)
```

Comment:
```suggestion
        return GET(url, headers)
```
pass headers to set appropriate useragent and other headers to requests